pkg/roachprod/cloud/cluster_cloud.go

```diff
@@ -374,9 +413,26 @@ func GrowCluster(l *logger.Logger, c *Cluster, numNodes int) error {
 				c.Name, gce.ProviderName)
 		}
 	}
-	return vm.ForProvider(provider, func(p vm.Provider) error {
-		return p.Grow(l, c.VMs, c.Name, names)
+
+	var vmsLock syncutil.Mutex
+	err := vm.ForProvider(provider, func(p vm.Provider) error {
+		addedVms, err := p.Grow(l, c.VMs, c.Name, names)
+		if err != nil {
+			return err
+		}
+
+		// Update the list of VMs in the cluster.
+		vmsLock.Lock()
+		defer vmsLock.Unlock()
+		c.VMs = append(c.VMs, addedVms...)
```

Comment:
What if the added VMs break the canonical sort order?

Reply:
This actually cannot happen right now with the only provider that's implemented (`gce`), but you're right... better safe than sorry